### PR TITLE
Stax porting

### PR DESCRIPTION
**Pull request by sgliner-ledger**

### CODE REVIEW

**Comment by agrojean-ledger** [comment]
@sgliner-ledger did you take a look at the failed CI static analysis jobs ? Maybe some fixes are easy to implement.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ src/glyphs.h
 *.pyc
 .idea
 .vscode
+build
```
Comment (agrojean-ledger) [inline .gitignore:8]: You should be able to remove src/glyphs.h and .c from the .gitignore as it is under build directory now

**Comment by agrojean-ledger** [comment]
Can you add a version bump commit ? @sgliner-ledger

File: src/ui_nbgl.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "ui.h"
+#include "nbgl_use_case.h"
+
+typedef struct {
```
Comment (agrojean-ledger) [inline src/ui_nbgl.h]: If you don't need to expose `UiContext_t` outside of `ui_nbgl.c` maybe you can move the definition of this structure inside `ui_menu_nbgl.c` and remove this header file entirely ? This way you would have only the `ui.h` header and it makes the app easier to understand. Only one entry point into the ui API (both for nbgl and bagl versions)

File: src/ui_menu_nbgl.c [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  return true;
+}
+
+static void display_settings_menu(void);
+
+static void settings_control_cb(int token, uint8_t index) {
+  UNUSED(index);
+  switch (token) {
+  case SWITCH_PUBLIC_KEY_SET_TOKEN:
+    switches[0].initState = N_btchip.pubKeyRequestRestriction;
```
Comment (agrojean-ledger) [inline src/ui_menu_nbgl.c]: I think you don't need to set the switch init state here.

File: src/ui_menu_nbgl.c [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+static const char *const infoContents[] = {APPVERSION, "(c) 2022 Ledger"};
+
+static nbgl_layoutSwitch_t switches[NB_SETTINGS_SWITCHES];
+
+static void quit_cb(void) {
+    os_sched_exit(-1);
+}
+
+static void switch_public_key(void) {
+  uint8_t value = (N_btchip.pubKeyRequestRestriction ? 0 : 1);
```
Comment (agrojean-ledger) [inline src/ui_menu_nbgl.c]: uint8_t value = (N_btchip.pubKeyRequestRestriction != 0) ? 0 : 1; would be more readable and pubKeyRequestRestriction is not really a boolean

File: src/ui_nbgl.c [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+static void status_callback(bool confirm) {
+  if (confirm) {
+    approved_status();
+  } else {
+    abandon_status();
+  }
+}
+
+// Prompt Cancel
+static void prompt_cancel(uint8_t type) {
```
Comment (agrojean-ledger) [inline src/ui_nbgl.c]: You might want to define a type for your enum and update the prompt_cancel prototype, for instance : 

`static void prompt_cancel(PromptType prompt_type)`

File: src/ui_nbgl.c [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+  if (type == MESSAGE_TYPE) {
```
Comment (agrojean-ledger) [inline src/ui_nbgl.c]: Better to use a switch here.

File: src/ui_nbgl.c [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+static void message_review_callback(bool token) {
+  if (token) {
+    approved_status();
+  } else {
+    prompt_cancel_message();
+  }
+}
+
+// Continue Review
+static void continue_review(uint8_t type) {
```
Comment (agrojean-ledger) [inline src/ui_nbgl.c]: Same here, would be better with an enum type imo.

File: src/ui.h [hunk 8]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "ux.h"
+
+void ui_idle_flow(void);
+void ui_sign_flow(void);
```
Comment (agrojean-ledger) [inline src/ui.h]: The name `ui_sign_flow` seems a bit ambiguous to me. I would rather call it `ui_sign_message_flow` or `ui_sign_msg_flow` to make it more explicit.

File: src/ui_nbgl.c [hunk 9]
```diff
@@ -0,0 +1,433 @@
...
+    continue_review(MESSAGE_TYPE);
+}
+
+static void continue_transaction_review(void) {
+  continue_review(TRANSACTION_TYPE);
+}
+
+// UI Start
+static void ui_start(void (*cb)(void), uint8_t type) {
+  if (type == MESSAGE_TYPE) {
```
Comment (agrojean-ledger) [inline src/ui_nbgl.c]: I'd use a switch here as well (anywhere you're checking this message type thing) with an enum type.

File: src/ui_nbgl.c [hunk 10]
```diff
@@ -0,0 +1,433 @@
...
+        .quitToken = 0,
+        .tuneId = TUNE_TAP_CASUAL};
+
+    nbgl_pageContent_t content = {
+        .type = TAG_VALUE_LIST,
+        .tagValueList.nbPairs = uiContext.nbPairs,
+        .tagValueList.pairs = (nbgl_layoutTagValue_t *)uiContext.tagValues};
+    releaseContext();
+    pageContext =
+        nbgl_pageDrawGenericContent(&single_flow_callback, &info, &content);
```
Comment (agrojean-ledger) [inline src/ui_nbgl.c:297]: Why couldn't you use a static review here ? :thinking:

Comment (sgliner-ledger) [reply src/ui_nbgl.c:297]: I wanted to be able to specify the page count regarding the number of outputs to be displayed

Comment (agrojean-ledger) [reply src/ui_nbgl.c:297]: You could probably use [nbgl_useCaseGetNbPagesForTagValueList](https://github.com/LedgerHQ/ledger-secure-sdk/blob/bdb5220a7841d93dcbb83e35a7d128a31d60094f/lib_nbgl/include/nbgl_use_case.h#L92) from the SDK with the static review to achieve your goal.

**Review by agrojean-ledger — APPROVED** [review]

**Pull request merged**